### PR TITLE
exit with correct exit code on plugin failure

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Correctly exits after a plugin command execution failure with the correct exit code.
Related downstream bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1501756

**Before**
```
$ kubectl plugin will-fail-plugin
ls: cannot access 'does-not-exist': No such file or directory
error: exit status 2

$ echo $?
1
```

**After**
```
$ kubectl plugin will-fail-plugin
ls: cannot access 'does-not-exist': No such file or directory
error: exit status 2

$ echo $?
2
```

cc @fabianofranz

### CODE REVIEW

File: pkg/kubectl/cmd/plugin.go [hunk 1]
```diff
@@ -109,6 +112,15 @@ func NewCmdForPlugin(f cmdutil.Factory, plugin *plugins.Plugin, runner plugins.P
 			}
 
 			if err := runner.Run(plugin, runningContext); err != nil {
+				if exiterr, ok := err.(*exec.ExitError); ok {
+					// check for (and exit with) the correct exit code
+					// from a failed plugin command execution
+					if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
```
Comment (fabianofranz) [inline pkg/kubectl/cmd/plugin.go]: Hrm, need to check if this works on Mac and Windows.

Comment (juanvallejo) [reply pkg/kubectl/cmd/plugin.go]: Sure, will go ahead and test this on both

Comment (juanvallejo) [reply pkg/kubectl/cmd/plugin.go]: Seems to work as expected on Windows:
windows_exit_code

Will test on mac tomorrow and post screenshot for it.

Comment (juanvallejo) [reply pkg/kubectl/cmd/plugin.go]: @fabianofranz works as expected on mac:
screen shot 2017-10-20 at 5 05 13 pm

@sg00dwin thanks for helping me test on mac

**Pull request merged**